torch/distributed/distributed_c10d.py

```diff
@@ -2380,23 +2409,17 @@ def recv(
     if tensor.is_complex():
         tensor = torch.view_as_real(tensor)
 
-    pg = group or _get_default_group()
+    group = _group_or_default_group(group)
 
-    if src is None:
-        work = pg.recv_anysource([tensor], tag)
+    if src is None and group_src is None:
+        work = group.recv_anysource([tensor], tag)
         work.wait()
         src_rank = work._source_rank()
-        if group is None or group is GroupMember.WORLD:
-            return src_rank
-        else:
-            return get_global_rank(pg, src_rank)
+        return get_global_rank(group, src_rank)
     else:
-        if group is None or group is GroupMember.WORLD:
-            pg.recv([tensor], src, tag).wait()
-        else:
-            group_src_rank = get_group_rank(pg, src)
-            pg.recv([tensor], group_src_rank, tag).wait()
-        return src
+        group_src = _canonicalize_group_rank(group, src, group_src)
```

Comment:
Can we use `_check_not_self_rank(group, group_src, "source")` here?

Reply:
hmm, no, we apparently can't.  It causes this test to fail 
` python test/distributed/test_distributed_spawn.py TestDistBackendWithSpawn.test_batch_isend_irecv_nccl`

The test looks kinda dumb, i'm not sure if it really even intends to send/recv to itself, but that's what it does.  And I think (guess?) that nccl supports this so we probably shouldn't start asserting against it now...